### PR TITLE
Create PDFs for new Help Center

### PR DESCRIPTION
**Pull request by ghost**
fixes https://github.com/trailbehind/GaiaCloud/issues/3512

### CODE REVIEW

**Comment by torreyhog** [comment]
fixes: https://github.com/trailbehind/GaiaCloud/issues/8551

Previously using zendesk account `torrey@gaiagps.com`, which was deactivated several months ago. Switch to use `support+devsupport@gaiagps.com` instead. 

Fixed Beautifulsoup error preventing "href"s from being read.

Built and uploaded new docker image to ECR - https://us-east-1.console.aws.amazon.com/ecr/repositories/private/265147712162/zendesk-utils?region=us-east-1

Task Definiton - https://us-east-1.console.aws.amazon.com/ecs/home?region=us-east-1#/taskDefinitions/zendesk-pdf/5

Ran task - https://us-east-1.console.aws.amazon.com/ecs/home?region=us-east-1#/clusters/gaiagps-production/tasks/b3f6bf65bf924610b9cdec9f9525d17e/details

File: to_json/ZendeskJsonPackager.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def fetch_all_sections(self):
     else:
       print ("ZENDESK API: fetching sections")
 
-    url = '{}/sections.json'.format(self.zendesk_url)
+    url = '{}/sections.json'.format(self.zendesk_url) + '?page[size]=100'
```
Comment (melanieawilson) [inline to_json/ZendeskJsonPackager.py:58]: This is the fix for the original bug. Our current Zendesk Help Center has 32 sections by default the sections API endpoint only returns 30 items per page. This resulted in a KeyError. Bumping this to 100 for now but a long term fix should include pagination.

File: helpcenter_to_pdf/ZendeskPDFMaker.py [hunk 2]
```diff
@@ -306,6 +307,7 @@ def wkhtmltopdf_options(self):
             "margin-bottom": "0.75in",
             "margin-left": "0.75in",
             "footer-right": "[page]",
+            "enable-local-file-access": "",
```
Comment (melanieawilson) [inline helpcenter_to_pdf/ZendeskPDFMaker.py:310]: Including this option resolved a ProtocolUnknownError with pdfit/ wkhtmltopdf. It also allowed for access to the files in the data/images directory that are used to create the cover for the PDF.

File: helpcenter_to_pdf/ZendeskPDFMaker.py [hunk 3]
```diff
@@ -140,6 +140,7 @@ def create_manual(self, category_id, category_name, locale):
             toc=self.table_of_contents_dict(local_toc_dict[locale]),
             cover=self.path_for_created_cover(local_title_dict[locale]),
             options=self.wkhtmltopdf_options(),
+            cover_first=True,
```
Comment (melanieawilson) [inline helpcenter_to_pdf/ZendeskPDFMaker.py:143]: This option puts the cover before the TOC.

**Comment by melanieawilson** [comment]
Updates in the latest commit:

- Fixed a bug in `fetch_all_sections `that resolved a KeyError. The help center currently has 32 sections. The API endpoint only returns 30 sections per page. This change now pulls in up to 100 sections. Long term fix should include pagination.
- Added option to put cover first.
- Added enable-local-file-access to fix an ProtocolUnknownError with pdfkit/ wkhtmltopdf. This change also allowed access to the image files used to generate the cover.
- Updated and pinned the requirements. Upgrading Beautiful Soup resolved a `AttributeError: module 'collections' has no attribute 'Callable'` error.

These updates allowed me to successfully generate PDFs locally and post them to the s3 bucket by running these commands:

`python3 ./starter.py create`
`python3 ./starter.py post`

Future updates to be addressed in separate PRs:

- Pagination for all API calls to zendesk (sections, categories). Articles already includes pagination.
- Getting the test automatically rerunning on AWS. This will require updating the docker image. Getting this to run successfully in the docker image. Getting the image and task updated in AWS.

**Review by andharris — APPROVED** [review]

**Review by jaredmolson — APPROVED** [review]
The changes we worked on today are all here. There's a lot we didn't write and it could be improved in areas, but I think this is good for now.

**Pull request merged**